src/weather.cpp

```diff
@@ -74,7 +74,8 @@ void weather_effect::glare( bool snowglare )
     }
     if( is_player_outside() && !g->u.in_sleep_state() && season == WINTER &&
         !g->u.worn_with_flag( "SUN_GLASSES" ) && !g->u.is_blind() && snowglare == true &&
-        !g->u.has_bionic( bionic_id( "bio_sunglasses" ) ) ) {
+        !g->u.has_bionic( bionic_id( "bio_sunglasses" ) ) &&
+        !calendar( to_turn<int>( calendar::turn ) ).is_night() ) {
```

Comment:
The ```g->is_in_sunlight( g->u.pos() )``` from the conditional statement above appears to be better than `is_night() `since the former takes light level into account, as well as the actual presence of the sunlight.

I'd also suggest a minor cleanup here to deduplicate the conditionals and turn them into a guard clause like this:
```c++
if( !is_player_outside() || g->u.in_sleep_state() || !g->is_in_sunlight( g->u.pos() ) ...)
    return;

if( !snowglare ) {
    ...
} else if (season_of_year( calendar::turn ) == WINTER) {
    ...
}
```

Reply:
is_in_sunlight only returns true, if the weather is currently sunny, snowglare only happens when the weather is currently snowing, the two cannot happen at same time, hence the check for night-time.
I am currently making a new snow mechanic tht would track snowfall, and that wouldnt require the current weather to be snowing.
Until then however, the closest approximation to snow on the ground is that it is currently snowing and it is winter.
The cleanup I will do.

Reply:
> snowglare only happens when the weather is currently snowing, the two cannot happen at same time, hence the check for night-time.

There seems to be a contradiction here. The description reads: "The sunlight reflecting off the snow makes it hard to see". This implies direct sunlight from the clear sky and mounds of already fallen snow (as I see it). The snow which is still falling, on the other hand, usually implies overcast conditions (which prevents any glares).

Reply:
Then perhaps I should remove snowglare entirely until I finish the mechanic of snow falling on the ground and building up. 
Currently the presence of snow can only be indicated by the season and the current weather.
The current weather "snowing" can imply that some of it has settled on the ground, and there is enough sunlight to cause a glare.
Do you think I should remove the glare until the snowfall mechanic is added? when it is added then a check for sunny weather would make more sense with the glare.

Reply:
I would say there's no need for a distinct effect in the first place. The sun glare effect itself can become more severe (doubled?) in case of presence of the fallen snow: not only the Sun itself strikes one's eyes, but its reflection in the snow as well. I suppose the season based check would suffice until you replace it with more sophisticated check.

Reply:
I'll close this in favour of #28227  which only applies snowglare when its sunny and snow has settld on the ground